SYSTEM_TESTS.md

```diff
@@ -0,0 +1,349 @@
+ 
+
+# Guide on System Tests in Ruby
+
+System tests in Ruby are special types of tests.
+
+They allow us to interact with our web app, test the UX and JavaScript of our application.
+
+This repository uses [Capybara](https://github.com/teamcapybara/capybara) library to perform system tests. It simulates user's interaction with our web app (visiting pages, clicking buttons, filling forms...)
+
+System tests are the slowest type of tests and we need to be careful when writing them.
+
+
+
+Visit the official [Capybara Docs](https://www.rubydoc.info/github/jnicklas/capybara/Capybara) to learn more.
+
+
+
+## Test's file structure
+
+The example code below shows how the normal test file looks like and introduces you to the basics of system tests.
+
+```ruby
+# Import system test configuration (driver, options...)
+require "application_system_test_case"
+
+# Class name should be descriptive and align with the filename
+class EditorTest < ApplicationSystemTestCase
+    # The time Capybara should wait for a function to complete (visit, find, all...)
+    Capybara.default_max_wait_time = 60
+    # Other includes and settings
+    
+    def setup
+       # This function doesn't exist in every test file, it's optional.
+       # If we define it, it'll run before each test.
+       # For example, we want to login the user before each test.
+    end
+    
+    # The standard test structure
+    test 'creating 3x3 table in Markdown' do
+       # Code ...
+    end
+    
+    # Other tests
+end
+```
+
+
+
+## Basic functions
+
+#### **visit '/path'**
+
+`visit` navigate  to the specific URL. If the page doesn't load in certain period of time(timeout) the test fails.
+
+
+
+#### assert_equal 'expected_value', 'actual_value'
```

Comment:
This part is common to all tests in the default rails testing framework. Maybe we can remove it?

Reply:
That's why I put it in the 'Basic functions' sections.